src/VisualStudio/Core/Def/Implementation/FindReferences/Contexts/AbstractTableDataSourceFindUsagesContext.cs

```diff
@@ -85,15 +85,22 @@ private abstract class AbstractTableDataSourceFindUsagesContext :
 
             #endregion
 
+            public sealed override CancellationToken CancellationToken => _cancellationTokenSource.Token;
+
             protected AbstractTableDataSourceFindUsagesContext(
                  StreamingFindUsagesPresenter presenter,
                  IFindAllReferencesWindow findReferencesWindow,
                  ImmutableArray<ITableColumnDefinition> customColumns,
                  bool includeContainingTypeAndMemberColumns,
-                 bool includeKindColumn)
+                 bool includeKindColumn,
+                 CancellationToken cancellationToken)
             {
                 presenter.AssertIsForeground();
 
+                // Wrap the passed in CT with our own CTS that we can control cancellation over.  This way either our
+                // caller can cancel our work or we can cancel the work.
+                _cancellationTokenSource = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken);
```

Comment:
❔ Where is this disposed (failure to dispose a linked cancellation token source will leak memory)

Reply:
technically isn't.  but i can add that.  looks like a problem we had before.  weo only ever cancel this CTS, never dispose it.

Reply:
Disposed now in our centralized path where we cancel searching.  this is called when the window is clsoed, reused for new results, or the search completes.

Reply:
> looks like a problem we had before

Previously, we weren't using a _linked_ token source. Only sources that are linked or have an active timeout allocate resources that are released by `Dispose()`.